crates/edr_rpc_server/src/node.rs

```diff
@@ -437,4 +618,548 @@ mod tests {
+                .await
+                .unwrap();
+
+            assert_eq!(count, expected_count);
+        }
+
+        let fixture = NodeTestFixture::new().await?;
+
+        let block_and_total_difficulty = fixture
+            .node
+            .block_and_total_difficulty_by_block_spec(&BlockSpec::Tag(BlockTag::Earliest))
+            .await
+            .unwrap();
+
+        assert_block_transaction_count(&fixture, BlockSpec::Tag(BlockTag::Earliest), 0).await;
+        assert_block_transaction_count(&fixture, BlockSpec::Tag(BlockTag::Latest), 0).await;
+        assert_block_transaction_count(&fixture, BlockSpec::Tag(BlockTag::Pending), 0).await;
+        assert_block_transaction_count(&fixture, BlockSpec::Number(0), 0).await;
+
+        assert_block_transaction_count(
+            &fixture,
+            BlockSpec::Eip1898(Eip1898BlockSpec::Number { block_number: 0 }),
+            0,
+        )
+        .await;
+
+        assert_block_transaction_count(
+            &fixture,
+            BlockSpec::Eip1898(Eip1898BlockSpec::Hash {
+                block_hash: block_and_total_difficulty.block.header().hash(),
+                require_canonical: None,
+            }),
+            0,
+        )
+        .await;
+
+        let non_existing_count = fixture
+            .node
+            .block_transaction_count_by_block_spec(&BlockSpec::Number(1))
+            .await;
+
+        assert_error!(non_existing_count, NodeError::UnknownBlockNumber { block_number } => assert_eq!(block_number, 1));
+
+        let non_existing_count = fixture
+            .node
+            .block_transaction_count_by_block_spec(&BlockSpec::Eip1898(Eip1898BlockSpec::Number {
+                block_number: 1,
+            }))
+            .await;
+
+        assert_error!(non_existing_count, NodeError::UnknownBlockNumber { block_number } => assert_eq!(block_number, 1));
+
+        let non_existing_count = fixture
+            .node
+            .block_transaction_count_by_block_spec(&BlockSpec::Eip1898(Eip1898BlockSpec::Hash {
+                block_hash: B256::zero(),
+                require_canonical: None,
+            }))
+            .await;
+
+        assert_error!(non_existing_count, NodeError::UnknownBlockHash { block_hash } => assert_eq!(block_hash, B256::zero()));
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn transaction_by_block_hash_and_index() -> Result<()> {
+        let fixture = NodeTestFixture::new().await?;
+
+        let block_and_total_difficulty = fixture
+            .node
+            .block_and_total_difficulty_by_block_spec(&BlockSpec::Tag(BlockTag::Latest))
+            .await?;
+
+        let tx = fixture
+            .node
+            .transaction_by_block_hash_and_index(
+                &block_and_total_difficulty.block.header().hash(),
+                0,
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(tx, None);
+
+        let non_existing_block_tx = fixture
```

Comment:
This seems like a separate test. I'd make two tests, named to indicate what they're testing. When a test fails, it makes it easier to understand which feature/functionality is broken. It also allows for parallel execution